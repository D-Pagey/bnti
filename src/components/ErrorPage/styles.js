import styled from 'styled-components';
import { padding } from 'polished';
import { colour, fontSize, spacing } from '../../tokens';

export const Background = styled.div`
    background-color: ${colour.black};
    height: 100vh;
    ${padding(spacing.big, spacing.medium)}
`;

export const Wrapper = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    max-width: 375px;
`;

export const Title = styled.h1`
    color: ${colour.blue};
    font-size: ${fontSize.big};
`;

export const Image = styled.img`
    height: auto;
    width: 150px;
`;

export const Text = styled.p`
    color: ${colour.blue};
    font-size: ${fontSize.medium};
    width: 100%;
`;
