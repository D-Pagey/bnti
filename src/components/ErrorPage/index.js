import React from 'react';
import pixelMan from './assets/pixel-man.png';
import * as S from './styles';

const ErrorPage = () => (
    <S.Background>
        <S.Wrapper>
            <S.Title>Oops something went wrong!</S.Title>
            <S.Image src={pixelMan} alt="pixel man" />
            <S.Text>Turn around! The circle is the other way!</S.Text>
            <S.Text>Theres nothing here, no loot, no bounties, nothing, nada.</S.Text>
        </S.Wrapper>
    </S.Background>
);

export default ErrorPage;
