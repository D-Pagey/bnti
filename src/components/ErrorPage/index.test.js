import React from 'react';
import ErrorPage from '.';

describe('ErrorPage component', () => {
    it('should render', () => {
        const { container } = render(<ErrorPage />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
