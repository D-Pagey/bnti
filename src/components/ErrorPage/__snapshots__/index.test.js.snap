// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage component should render 1`] = `
.c0 {
  background-color: #353535;
  height: 100vh;
  padding-top: 2rem;
  padding-right: 1rem;
  padding-bottom: 2rem;
  padding-left: 1rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 auto;
  max-width: 375px;
}

.c2 {
  color: #00A7CB;
  font-size: 2rem;
}

.c3 {
  height: auto;
  width: 150px;
}

.c4 {
  color: #00A7CB;
  font-size: 1.25rem;
  width: 100%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    >
      Oops something went wrong!
    </h1>
    <img
      alt="pixel man"
      class="c3"
      src="pixel-man.png"
    />
    <p
      class="c4"
    >
      Turn around! The circle is the other way!
    </p>
    <p
      class="c4"
    >
      Theres nothing here, no loot, no bounties, nothing, nada.
    </p>
  </div>
</div>
`;
